app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+
+            return ['uuid' => $uuid, 'id' => $id, 'area' => $areaHectares];
+        } catch (\Exception $e) {
+            echo $e;
+
+            return $e->getMessage();
+        }
+    }
+
+    private function validatePolygonBounds(array $geometry): bool
+    {
+        if ($geometry['type'] !== 'Polygon') {
+            return false;
+        }
+        $coordinates = $geometry['coordinates'][0];
+        foreach ($coordinates as $coordinate) {
+            $latitude = $coordinate[1];
+            $longitude = $coordinate[0];
+
+            // Check latitude bounds
+            if ($latitude < -90 || $latitude > 90) {
+                return false;
+            }
+
+            // Check longitude bounds
+            if ($longitude < -180 || $longitude > 180) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public function insertGeojsonToDB(string $geojsonFilename)
+    {
+        $srid = 4326;
+        $geojsonData = Storage::get("public/geojson_files/{$geojsonFilename}");
+        $geojson = json_decode($geojsonData, true);
+        if (! isset($geojson['features'])) {
+            return ['error' => 'GeoJSON file does not contain features'];
+        }
+        $uuids = [];
+        foreach ($geojson['features'] as $feature) {
+            if ($feature['geometry']['type'] === 'Polygon') {
+                if (! $this->validatePolygonBounds($feature['geometry'])) {
+                    return ['error' => 'Invalid polygon bounds'];
+                }
+                $data = $this->insertSinglePolygon($feature['geometry'], $srid);
+                $uuids[] = $data['uuid'];
+                $returnSite = $this->insertSitePolygon($data['uuid'], $feature['properties'], $data['area']);
+            } elseif ($feature['geometry']['type'] === 'MultiPolygon') {
+                foreach ($feature['geometry']['coordinates'] as $polygon) {
+                    if (! $this->validatePolygonBounds($feature['geometry'])) {
+                        return ['error' => 'Invalid polygon bounds'];
+                    }
+                    $data = $this->insertSinglePolygon(['type' => 'Polygon', 'coordinates' => $polygon], $srid);
+                    $uuids[] = $data['uuid'];
+                    $returnSite = $this->insertSitePolygon($data['uuid'], $feature['properties'], $data['area']);
+                }
+            }
+        }
+
+        return $uuids;
+    }
+
+    private function validateSchema(array $properties, array $fields): bool
+    {
+        foreach ($fields as $field) {
+            if (! array_key_exists($field, $properties)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public function validateDataInDB(Request $request)
+    {
+        $polygonUuid = $request->input('uuid');
+        $fieldsToValidate = ['poly_name', 'plantstart', 'plantend', 'practice', 'target_sys', 'distr', 'num_trees'];
+        $DATA_CRITERIA_ID = 14;
```

Comment:
What's the purpose of this magic number?

Reply:
I haven't dug in enough to tell for myself yet; does the table need to be in the DB? Is a join against these ever done at the SQL level? Or could this instead be in `config`, and the associate made at the code level?

Reply:
This will be used to join with other tables, just don't yet.